libs/ledger-live-common/src/families/cosmos/js-buildTransaction.ts

```diff
@@ -255,180 +251,21 @@ export const buildTransaction = async (
-            validatorAddress: transaction.validators[0].address,
-            amount: {
-              denom: account.currency.units[1].code,
-              amount: transaction.validators[0].amount.toString(),
-            },
-          },
-        });
-      }
-      break;
-
-    case "redelegate":
-      if (
-        !transaction.sourceValidator ||
-        !transaction.validators ||
-        transaction.validators.length < 1 ||
-        !transaction.validators[0].address ||
-        transaction.validators[0].amount.lte(0)
-      ) {
-        isComplete = false;
-      } else {
-        messages.push({
-          typeUrl: "/cosmos.staking.v1beta1.MsgBeginRedelegate",
-          value: {
-            validatorSrcAddress: transaction.sourceValidator,
-            delegatorAddress: account.freshAddress,
-            validatorDstAddress: transaction.validators[0].address,
-            amount: {
-              denom: account.currency.units[1].code,
-              amount: transaction.validators[0].amount.toString(),
-            },
-          },
-        });
-      }
-      break;
-
-    case "claimReward":
-      if (
-        !transaction.validators ||
-        transaction.validators.length < 1 ||
-        !transaction.validators[0].address
-      ) {
-        isComplete = false;
-      } else {
-        messages.push({
-          typeUrl: "/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward",
-          value: {
-            delegatorAddress: account.freshAddress,
-            validatorAddress: transaction.validators[0].address,
-          },
-        });
-      }
-      break;
-
-    case "claimRewardCompound":
-      if (
-        !transaction.validators ||
-        transaction.validators.length < 1 ||
-        !transaction.validators[0].address ||
-        transaction.validators[0].amount.lte(0)
-      ) {
-        isComplete = false;
-      } else {
-        messages.push({
-          typeUrl: "/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward",
-          value: {
-            delegatorAddress: account.freshAddress,
-            validatorAddress: transaction.validators[0].address,
-          },
-        });
-
-        messages.push({
-          typeUrl: "/cosmos.staking.v1beta1.MsgDelegate",
-          value: {
-            delegatorAddress: account.freshAddress,
-            validatorAddress: transaction.validators[0].address,
-            amount: {
-              denom: account.currency.units[1].code,
-              amount: transaction.validators[0].amount.toString(),
-            },
-          },
-        });
-      }
-      break;
-  }
-
-  if (!isComplete) {
-    return [];
-  }
-
-  return messages;
-};
-
-export const postBuildTransaction = async (
-  signResponse: AminoSignResponse,
+export const buildTransaction = async (
   protoMsgs: Array<ProtoMsg>,
+  memo: string,
+  pubKeyType: string,
+  pubKey: string,
+  feeAmount: any,
+  gasLimit: any,
+  sequence: string,
+  signature: Uint8Array,
 ): Promise<Uint8Array> => {
```

Comment:
Just a good practice: when starting to have a lot of params it's better to use a single object as param, so that each field is handled explicitely.
```suggestion
export const buildTransaction = async ({
  protoMsgs,
  memo,
  pubKeyType,
  pubKey,
  feeAmount,
  gasLimit,
  sequence,
  signature,
}:{
  protoMsgs: Array<ProtoMsg>;
  memo: string;
  pubKeyType: string;
  pubKey: string;
  feeAmount: any;
  gasLimit: any;
  sequence: string;
  signature: Uint8Array;
}): Promise<Uint8Array> => {
```

Reply:
Done